### PR TITLE
 Intel AMX accelerated Multicloud GitOps pattern with OpenShift AI - documentation

### PR DESCRIPTION
**Pull request by iirzynska**
Documentation draft for validatedpatterns.io

### CODE REVIEW

**Review by pgrabusz — CHANGES_REQUESTED** [review]
Great job Iza :) please check my in-line comments + here's the cluster sizing to change current values:

In the Cluster sizing tab, please change recommended cluster size:
* Control Planes CPU: 2x 5th Gen Intel® Xeon® Gold 6526Y (16 cores at 2.8 GHz base with Intel AMX) or better
* Workers CPU: 2x 5th Gen Intel® Xeon® Gold 6538Y+(32 cores at 2.2 GHz base with Intel AMX) or better

In the Cluster sizing tab, please change the "tested in the on-premises environment with following HW" table to:

* CP + W CPU: 2x 5th Gen Intel® Xeon® Platinum 8568Y+ (48 cores at 2.3 GHz base with Intel AMX)
* CP + W Memory: 512 GB (16x32GB DDR5 5600)

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+image::multicloud-gitops-amx-rhoai/amx-rhoai-validated-pattern-operator-2.png[Install Validated Patterns Operator,scale=50]
++
+
+. Select *Operators > Installed Operators*.
+
+. Ensure that *Validated Patterns Operator* is listed in the `openshift-operators` project with a status `Succeeded`.
+
+== Create {amx-rhoai-mcg-pattern}
+
+After succeeded installation open *Validated Patterns Operator*, go to *Pattern* tab and click *Create Pattern*.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc:154]: One more change request here:
After a successful installation, open the Validated Patterns Operator page. Next, go to the Pattern tab and click Create Pattern.

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-demo-script.adoc [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+----
++
+
+. After login go to `secret` catalog and clik *Create secret* and fill all the fields manually (Figure 2):
+
+.. *Path for this secret* is `global/config-demo` (from _values.yaml_ file for `config-demo` charts)
+
+.. Under *Secret data* key is `secret` (from _values-secret-multicloud-gitops.yaml_ file) and in next field put its value.
+
+.. Click *Add* and then *Save*.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-demo-script.adoc:99]: One more comment...

TBH I don't get it... Let me try to rephrase it, and decide if this is clear:

a) Put `global/config-demo` value in the _Path for this secret_ field (the value cames from _values-secret-multicloud-gitops.yaml_ file).
b) Add one _Secret data_ key-value pair. Put `secret` as a key (left field). The value must acquired from _values-secret-multicloud-gitops.yaml_ file. Click _Add_ button to confirm.
c) Click _Save_ to save changes.

THIS IS HOW I UNDERSTAND THIS PART - PLEASE VALIDATE A LOGIC BEFORE REPLACING YOUR PART ;)

**BTW is this section twice? In the "Getting started" page, and in the "Demo Script"?**

File: content/patterns/multicloud-gitops-amx-rhoai/_index.adoc [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+* OpenVINO Toolkit Operator manages OpenVINO components within Openshift environment. First one, OpenVINO™ Model Server (OVMS) is a scalable, high-performance solution for serving machine learning models optimized for Intel® architectures. The other component, that was used in the proposed pattern is Notebook resource. This element integrates Jupyter from OpenShift AI with a container image that includes developer tools from the OpenVINO toolkit. It also enables selecting a defined image OpenVINO™ Toolkit from the Jupyter Spawner choice list.
+
+As an example of AI workload in the pattern serves BERT-Large model. It is a wide known model used by various enterprise Natural Language Processing workloads. Intel has demonstrated, that *{intel-5th-gen-xeon-processors}* perform up to *1.49* times better in NLP flows vs. prior generation:
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/_index.adoc:117]: I changed this part (with changes for next comment):
```
BERT-Large model is used as an example of AI workload using {intel-amx} in the pattern. The BERT-Large inference is running in the Jupyther Notebook that uses OpenVINO optimizations.

As a side note, BERT_Large is a wide known model used by various enterprise Natural Language Processing workloads. Intel has demonstrated, that *{intel-5th-gen-xeon-processors}* perform up to *1.49* times better in NLP flows on Red Hat OpenShift vs. prior generation of processors- read more:
https://community.intel.com/t5/Blogs/Tech-Innovation/Data-Center/Level-Up-Your-NLP-applications-on-Red-Hat-OpenShift-and-5th-Gen/post/1572320[Level Up Your NLP applications on Red Hat OpenShift and 5th Gen]
```

File: content/patterns/multicloud-gitops-amx-rhoai/_index.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+https://community.intel.com/t5/Blogs/Tech-Innovation/Data-Center/Level-Up-Your-NLP-applications-on-Red-Hat-OpenShift-and-5th-Gen/post/1572320[Level-Up-Your-NLP-applications-on-Red-Hat-OpenShift-and-5th-Gen]
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/_index.adoc:118]: answer in previous comment

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+[id="deploying-mcg-pattern"]
+== Deploying the {amx-rhoai-mcg-pattern}
+
+.Prerequisites
+
+* An OpenShift cluster
+ ** To create an OpenShift cluster, go to the https://console.redhat.com/[Red Hat Hybrid Cloud console] and select *Services \-> Containers \-> Create cluster*.
+ ** The cluster must have a dynamic `StorageClass` to provision `PersistentVolumes`.
+* The cluster must have worker nodes with 5th Generation Intel Xeon Scalable Processors
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc:21]: :) done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+== Create {amx-rhoai-mcg-pattern}
+
+After succeeded installation open *Validated Patterns Operator*, go to *Pattern* tab and click *Create Pattern*.
+
+. Fill the *Name* of the pattern `multicloud-gitops-amx-rhoai` and *Cluster Group Name* `hub` (from _values-global.yaml_ file).
+
+. Under *Git Config > Target Repo* copy the link to your fork and under *Git Config > Target Revision* write the name of your branch (Figure 3).
+
+. Click *Create* to create the pattern.
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc:160]: ok, done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-demo-script.adoc [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
++
+.Create secret
+image::multicloud-gitops-amx-rhoai/amx-rhoai-secret-vault.png[Create secret in the vault]
++
+
+[id="verify"]
+
+=== Verify the test web pages
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-demo-script.adoc:107]: done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+include::modules/comm-attributes.adoc[]
+:toc:
+:imagesdir: /images
+:_content-type: REFERENCE
+
+[id="demo-intro"]
+
+== Introduction
+The {amx-rhoai-mcg-pattern} provides tools for developers or data scientists to ...
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc:15]: I lost interest in this phrase and I ...

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+== AMX demo
+
+Using {intel-5th-gen-xeon-processors} the kernel detects Intel® AMX at run-time, there is no need to enable and configure it additionally to improve performance. However, we need Intel optimized tools and frameworks to take advantage of AMX acceleration, such as OpenVINO Toolkit. 
+
+Before proceeding with this demo steps, please make sure you have your pattern deployed with link:../mcg-amx-rhoai-getting-started[Getting started]
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc:21]: done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+. Download BERT-large example, that uses AMX accelerator, by typing in the opened notebook:
+[source,terminal]
+!wget https://raw.githubusercontent.com/iirzynska/multicloud-gitops-ii/intel-rhods-amx/scripts/BERT.ipynb -O BERT.ipynb
+
+// change later for proper link something like:
+// !wget https://raw.githubusercontent.com/validatedpatterns-sandbox/amx-accelerated-rhoai-multicloud-gitops/scripts/BERT.ipynb -O BERT.ipynb
+
+
+. On the left-hand side menu the `BERT.ipynb` script should show up. Open it and run instructions one by one with play button or with `Ctr+Enter` from keyboard
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc:40]: 👍

[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc:40]: 👍

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+In ONEDNN verbose you should see *`avx_512_core_amx`* entry, what confirms that AMX instructions are being used.
+
+.BERT inference log
+image::multicloud-gitops-amx-rhoai/amx-rhoai-bert-logs.png[Logs from amx-app pod]
+
+To compare results without AMX you may run this benchmark with different precision i.e. FP32.
+[source,terminal]
+----
+!benchmark_app -m bert-large-uncased-whole-word-masking-squad-0001.xml -infer_precision f32
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-bert-script.adoc:74]: done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-schematic-diagrams.adoc [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+include::modules/comm-attributes.adoc[]
+
+:toc:
+:imagesdir: /images
+:_content-type: ASSEMBLY
+
+
+[id="overview-architecture"]
+== Overview of the architectures
```
[the remark this answers is not in the document]
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-schematic-diagrams.adoc:15]: done

File: content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc (hunk 1 above)
Comment (iirzynska) [reply content/patterns/multicloud-gitops-amx-rhoai/mcg-amx-rhoai-getting-started.adoc:154]: done

**Pull request merged**